### DISCUSSION TITLE
Equivalent to setSetting event

### DISCUSSION THREAD

**Discussion by ollebergkvist** [ai/nanostores#123] [Q&A]
I need to trigger useEffect when the value e.g setSetting is run, not when the value has been set eg. setting?
How can one do that with nanostores?

**Comment by ai** [comment]
> the value e.g setSetting is run

What does `run` mean?

**Reply by ollebergkvist** [reply]
Whenever the setSetting is triggered.

So if we take React as an example, you could use "count" or "setCount" as your useEffect dependency.
const [count, setCount] = useState(0);

I'm not sure how to do this the nanostores way?

**Reply by ai** [reply]
You can use `onSet` event. It triggers before store will change the value (you can even abort changes).

```js
import { onSet } from 'nanostores'

onSet(store, e => {
})
```
